v2/tools/generator/internal/codegen/pipeline/load_types.go

```diff
@@ -94,7 +94,7 @@ func LoadTypes(
 
 				scope := categorizeResourceScope(resourceInfo.ARMURI)
 				resourceType = resourceType.WithScope(scope)
-				resourceDefinition := astmodel.MakeTypeDefinition(resourceName, resourceType)
+				resourceDefinition := astmodel.MakeTypeDefinition(resourceName.(astmodel.InternalTypeName), resourceType)
```

Comment:
Will improve later? icky atm.

Reply:
Yup. There's a lot of this that will go away.